providers/hubspot/record.go

```diff
@@ -0,0 +1,55 @@
+package hubspot
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path"
+
+	"github.com/amp-labs/connectors/common"
+)
+
+/*
+   docs:
+   https://developers.hubspot.com/beta-docs/reference/api/crm/objects/companies
+   https://developers.hubspot.com/beta-docs/reference/api/crm/objects/contacts
+   https://developers.hubspot.com/beta-docs/reference/api/crm/objects/deals
+   https://developers.hubspot.com/beta-docs/reference/api/crm/objects/tickets
+   https://developers.hubspot.com/beta-docs/reference/api/crm/objects/line_items
+   https://developers.hubspot.com/beta-docs/reference/api/crm/objects/products
+*/
+
+//nolint:gochecknoglobals
+var getRecordSupportedObjectsToPathMap = map[string]string{
+	"company":   "companies",
+	"contact":   "contacts",
+	"deal":      "deals",
+	"ticket":    "tickets",
+	"line_item": "line_items",
+	"product":   "products",
+}
+
+var ErrUnsupportedObject = errors.New("unsupported object")
+
+func (c *Connector) GetRecord(ctx context.Context, objectName string, recordId string) (*common.ReadResultRow, error) {
```

Comment:
What is the meaning of `GetRecord` method?

When should we use `Search` and when `GetRecord`. Is `GetRecord` more general or relevant in the context of Webhook?

Reply:
Ok, I will think of it as "Singular Read" on demand